jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/exec/EcsExecUtils.kt

```diff
@@ -95,12 +103,76 @@ object EcsExecUtils : CoroutineScope by ApplicationThreadPoolScope("EcsExec") {
+            val containerInstanceArn = task.containerInstanceArn()
+            val res = ecsClient.describeContainerInstances {
+                it.cluster(clusterArn)
+                it.containerInstances(containerInstanceArn)
+            }
+            val ec2InstanceId = res.containerInstances().first().ec2InstanceId()
+            val instanceProfileArn = project.awsClient<Ec2Client>().describeInstances {
+                it.instanceIds(ec2InstanceId)
+            }.reservations().first().instances().first().iamInstanceProfile().arn() ?: return null
+            val instanceProfileName = instanceProfileArn.substringAfter(":instance-profile/")
+            return iamClient.getInstanceProfile { it.instanceProfileName(instanceProfileName) }.instanceProfile().roles().first().arn() ?: null
+        } catch (e: Exception) {
+            return null
+        }
+    }
+
+    fun getTaskRoleArn(project: Project, clusterArn: String, taskArn: String): String? {
+        val ecsClient = project.awsClient<EcsClient>()
+        val task = ecsClient.describeTasks {
+            it.tasks(taskArn)
+            it.cluster(clusterArn)
+        }.tasks().first()
+        return if (task.overrides().taskRoleArn() != null) {
+            task.overrides().taskRoleArn()
+        } else {
+            project.getResourceNow(EcsResources.describeTaskDefinition(task.taskDefinitionArn())).taskRoleArn()
+                ?: when (task.launchType()) {
+                    LaunchType.EC2 -> getEc2InstanceTaskRoleArn(project, clusterArn, ecsClient, task)
+                    LaunchType.FARGATE -> null
+                    else -> throw RuntimeException("Launch Type is not supported")
+                }
+        }
+    }
+
+    fun checkRequiredPermissions(project: Project, clusterArn: String, taskArn: String): Boolean {
+        try {
+            val iamClient = project.awsClient<IamClient>()
+            val taskRoleArn = getTaskRoleArn(project, clusterArn, taskArn) ?: return false
+
+            val permissions = listOf(
+                SESSION_MANAGER_CREATE_CONTROL_CHANNEL_PERMISSION,
+                SESSION_MANAGER_CREATE_DATA_CHANNEL_PERMISSION,
+                SESSION_MANAGER_OPEN_CONTROL_CHANNEL_PERMISSION,
+                SESSION_MANAGER_OPEN_DATA_CHANNEL_PERMISSION
+            )
+            val response = iamClient.simulatePrincipalPolicy {
+                it.policySourceArn(taskRoleArn)
+                it.actionNames(permissions)
+            }
+
+            val permissionResults = response.evaluationResults().map { it.evalDecision().name }
+            for (permission in permissionResults) {
+                if (permission != PolicyEvaluationDecisionType.ALLOWED.name) {
+                    return false
+                }
+            }
+        } catch (e: Exception) {
+            notifyWarn(message("ecs.execute_command_permissions_required_title"), message("ecs.execute_command_permissions_not_verified"))
```

Comment:
project should be passed along
```suggestion
            notifyWarn(
                project = project,
                title = message("ecs.execute_command_permissions_required_title"),
                content = message("ecs.execute_command_permissions_not_verified")
            )
```

Reply:
Good catch

Reply:
Added the parameter names